### PR TITLE
22224-check-fields-changing

### PR DESCRIPTION
**Pull request by VVyacheslav**

### CODE REVIEW

File: admin_smoke/tests.py [hunk 1]
```diff
@@ -277,6 +272,19 @@ def test_changeform_create(self) -> None:
...
             self.assertEqual(obj.created, self.now)  # type: ignore
 
+    def test_all_fields_present(self) -> None:
+        """ All not excluded fields are present on the form. """
+        r = self.client.get(self.change_url)
+        cd = getattr(r, 'context_data')
+        form: ModelForm = cd['adminform'].form
+        model_fields = {f.name for f in self.opts.fields
+                        if f.name not in self.excluded_fields
+                        and not isinstance(f, AutoField)}
```
Comment (tumb1er) [inline admin_smoke/tests.py:282]: there should be a check for primary key, because it is not always AutoField

File: admin_smoke/tests.py [hunk 2]
```diff
@@ -277,6 +272,19 @@ def test_changeform_create(self) -> None:
...
+        form_fields = set(form.Meta.fields)
+        absent_fields = model_fields - form_fields
+        self.assertFalse(absent_fields,
```
Comment (tumb1er) [inline admin_smoke/tests.py:285]: assertSetEqual may be more appropriate here

File: testproject/testapp/models.py [hunk 3]
```diff
@@ -12,9 +12,15 @@ class Tag(models.Model):
...
 
+class Client(models.Model):
+    name = models.CharField(max_length=10, unique=True)
+
+
 class Project(models.Model):
     pid = models.AutoField(primary_key=True)
     name = models.CharField(max_length=10, unique=True)
+    client = models.ForeignKey(Client, on_delete=models.CASCADE, null=True,
+                               blank=True)
```
Comment (tumb1er) [inline testproject/testapp/models.py:23]: This field is visible in admin. Why is it excluded in corresponding AdminTestCase?

File: admin_smoke/tests.py [hunk 4]
```diff
@@ -30,6 +30,7 @@ class AdminBaseTestCase(BaseTestCase):
     changelist_url: str
     add_url: str
     admin: ModelAdmin
+    excluded_fields: List[str] = []
```
Comment (tumb1er) [inline admin_smoke/tests.py:33]: There should be some words about it in readme

File: admin_smoke/tests.py (hunk 2 above)
Comment (VVyacheslav) [reply admin_smoke/tests.py:285]: There can be additional fields on the form absent in model. Then assertSetEqual  is going to call a false fail.

File: admin_smoke/tests.py (hunk 1 above)
Comment (VVyacheslav) [reply admin_smoke/tests.py]: ++

File: testproject/testapp/models.py (hunk 3 above)
Comment (VVyacheslav) [reply testproject/testapp/models.py:23]: The field were not visible in admin. Something went wrong...  ++

File: admin_smoke/tests.py (hunk 4 above)
Comment (VVyacheslav) [reply admin_smoke/tests.py]: ++

File: admin_smoke/tests.py [hunk 5]
```diff
@@ -277,6 +272,20 @@ def test_changeform_create(self) -> None:
...
             self.assertEqual(obj.created, self.now)  # type: ignore
 
+    def test_all_fields_present(self) -> None:
+        """ All not excluded fields are present on the form. """
+        r = self.client.get(self.change_url)
+        cd = getattr(r, 'context_data')
+        form: ModelForm = cd['adminform'].form
+        model_fields = {f.name for f in self.opts.fields
+                        if f.name not in self.excluded_fields
+                        and not isinstance(f, AutoField)
```
Comment (tumb1er) [inline admin_smoke/tests.py:282]: f.primary_key check is enough.

Comment (VVyacheslav) [reply admin_smoke/tests.py]: ++

**Pull request merged**